prov/gni/src/gnix_msg.c

```diff
@@ -373,16 +374,19 @@ static int __gnix_rndzv_req_complete(void *arg, gni_return_t tx_status)
 
 	if (tx_status != GNI_RC_SUCCESS) {
 		req->tx_failures++;
-		if (req->tx_failures <
-		    req->gnix_ep->domain->params.max_retransmits) {
+		if (GNIX_EP_RDM(req->gnix_ep->type) &&
+			req->tx_failures <
+				req->gnix_ep->domain->params.max_retransmits) {
 
 			GNIX_INFO(FI_LOG_EP_DATA,
 				  "Requeueing failed request: %p\n", req);
 			return _gnix_vc_queue_work_req(req);
 		}
 
-		/* TODO should this be fatal? A request will sit waiting at the
-		 * peer. */
+		GNIX_INFO(FI_LOG_EP_DATA, "Dropping failed request: %p\n", req);
```

Comment:
Should this message different based on EP type?  If just to identify the EP type.  Or maybe it's not even of interest in the DGRAM case.